packages/stargate/src/aminotypes.spec.ts

```diff
@@ -353,6 +353,70 @@ describe("AminoTypes", () => {
       expect(aminoMsg).toEqual(expected);
     });
 
+    it("works for MsgTransfer with empty values", () => {
+      const msg: MsgTransfer = {
+        sourcePort: "testport",
+        sourceChannel: "testchannel",
+        token: coin(1234, "utest"),
+        sender: "cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6",
+        receiver: "cosmos10dyr9899g6t0pelew4nvf4j5c3jcgv0r73qga5",
+        timeoutHeight: {
+          revisionHeight: Long.UZERO,
+          revisionNumber: Long.UZERO,
+        },
+        timeoutTimestamp: Long.UZERO,
+      };
+      const aminoMsg = new AminoTypes().toAmino({
+        typeUrl: "/ibc.applications.transfer.v1.MsgTransfer",
+        value: msg,
+      });
+      const expected: AminoMsgTransfer = {
+        type: "cosmos-sdk/MsgTransfer",
+        value: {
+          source_port: "testport",
+          source_channel: "testchannel",
+          token: coin(1234, "utest"),
+          sender: "cosmos1pkptre7fdkl6gfrzlesjjvhxhlc3r4gmmk8rs6",
+          receiver: "cosmos10dyr9899g6t0pelew4nvf4j5c3jcgv0r73qga5",
+          timeout_height: {
```

Comment:
Is there a difference between this and `timeout_height: {}`?

I guess JSON (at least) would encode them the same. Is there some other subtle difference that might appear in JS land?

Reply:
In JS and Jasmine testing there is a difference between `{ foo: undefined }` and `{}`. However, when serializing them to JSON they become the same.

It is written that way because the implementation always sets the `revision_height`/`revision_number` properties. We could change that but this would create a lot of hard to read code.